lib/features/docker_exec_websocket_server.js

```diff
@@ -0,0 +1,80 @@
+const MAX_RANDOM_PORT_ATTEMPTS = 20;
+
+var debug = require('debug')('docker-worker:features:docker_exec_websocket_server.js');
+var DockerExecServer = require('docker-exec-websocket-server').DockerExecServer;
+// var waitForPort = require('../wait_for_port');
+var http = require('http');
+var slugid = require('slugid');
+var Promise = require('promise');
+var url = require('url');
+
+export default class WebsocketServer {
+  constructor () {
+    this.path = '/' + slugid.v4();
+    // this.path = '/a';
+  }
+
+  async started (task) {
+    debug('creating ws server');
+    //TODO: add https support
+    var httpServ = http.createServer();
+    var port;
+    //searching for an open port between 32768 and 61000
+    var attempts = 0;
+    while (true) {
+      // port = 40836;
+      port = Math.floor((Math.random() * (61000 - 32768)) + 32768); 
+      try {
+        await (new Promise((resolve, reject) => {
+          httpServ.listen(port);
+          httpServ.once('listening', () => {
+            httpServ.removeListener('error', reject);
+            resolve();
+          });
+          httpServ.once('error', reject)}));
+        break;
+      } catch (err) {
+        // Only handle address in use errors.
+        if (err.code !== 'EADDRINUSE') {
+          throw err;
+        }
+        attempts += 1;
+        if (attempts >= MAX_RANDOM_PORT_ATTEMPTS) {
+          throw err;
+        }
+      }
+    }
+    debug('port found', port);
+    this.server = new DockerExecServer({
+      server: httpServ,
+      containerId: task.dockerProcess.id,
+      path: this.path,
+    });
+    this.url = url.format({
+      protocol: 'ws',
```

Comment:
I was just talking to garndt, and the ssl certificate used for testing:
https://github.com/taskcluster/docker-worker/blob/docker-exec/config/test.js#L22-L25

Is signed for `localhost`, it's self-signed so not really worth much and won't validate. But if we set `rejectUnauthorized: false,` when we make the request node won't reject an unauthorized certificate. This is only something we should do in testing, but it's easily possible see:
https://github.com/websockets/ws/blob/master/doc/ws.md#new-wswebsocketaddress-protocols-options

We just need to allow the option in `DockerExecClient` and pass it on to `ws`. Note, it's important that `rejectUnauthorized` is `true` by default, ether `true` or not specified as a property on the options object given to `ws`. I'm not sure `rejectUnauthorized: undefined` is good enough.

------

Note, the ssl certificate filenames are stored in `task.runtime.ssl.certificate` and `task.runtime.ssl.key`, see: 
https://github.com/taskcluster/docker-worker/blob/docker-exec/lib/features/local_live_log.js#L75-L76
Note it seems we don't even need to modify `DockerExecServer` here, we just use `require('https')` and give it there, see:
https://github.com/websockets/ws/blob/master/examples/ssl.js#L34-L35